app/models/user.rb

```diff
@@ -470,6 +473,15 @@ def find_ids(options)
 
   protected
 
+  def promote_to_first_admin
+    account.update({first_admin_id: id})
+  end
+
+  def update_account_first_admin
+    return unless (other_admin = account.admins.where.not(id: id).first)
+    account.update!({first_admin_id: other_admin.id})
```

Comment:
`other_admin.promote_to_first_admin!`

Reply:
`promote_to_first_admin` Is a method in `User` that does:
```
def promote_to_first_admin
  account.update({first_admin_id: id})
end
```

But `other_admin` is an `Account`, not a `User`.
Do you want me to create a method `promote_to_first_admin!` sending the ID as a param then? What's the benefit? I know that we updating the account but since this is internal logic of the User that happens when the user is created/deleted, imo it is good that stays inside the User as private methods.